packages/grafana-ui/src/components/FileDropzone/FileDropzone.test.tsx

```diff
@@ -0,0 +1,149 @@
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import React, { ReactNode } from 'react';
+import { FileDropzone } from './FileDropzone';
+
+const file = ({
+  fileBits = JSON.stringify({ ping: true }),
+  fileName = 'ping.json',
+  options = { type: 'application/json' },
+}) => new File([fileBits], fileName, options);
+
+const files = [
+  file({}),
+  file({ fileName: 'pong.json' }),
+  file({ fileBits: 'something', fileName: 'something.jpg', options: { type: 'image/jpeg' } }),
+];
+
+describe('The FileDropzone component', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should show the default text of the dropzone component when no props passed', () => {
+    render(<FileDropzone />);
+
+    expect(screen.getByText('Upload file')).toBeInTheDocument();
+  });
+
+  it('should show accepted file type when passed in the options as a string', () => {
+    render(<FileDropzone options={{ accept: '.json' }} />);
+
+    expect(screen.getByText('Accepted file type: .json')).toBeInTheDocument();
+  });
+
+  it('should show accepted file types when passed in the options as a string array', () => {
+    render(<FileDropzone options={{ accept: ['.json', '.txt'] }} />);
+
+    expect(screen.getByText('Accepted file types: .json, .txt')).toBeInTheDocument();
+  });
+
+  it('should handle file removal from the list', async () => {
+    const { rerender, container } = render(<FileDropzone />);
+    const dropzone = container.querySelector('div[TabIndex="0"]');
+
+    dispatchEvt(dropzone, 'drop', mockData(files));
+
+    await flushPromises(rerender, <FileDropzone />);
```

Comment:
you shouldn't need to do this anywhere if you use the asynchronous `find` methods instead of the synchronous `get` ones. e.g. i think replace `expect(screen.getAllByLabelText('Remove')).toHaveLength(3);` with `expect(await screen.findAllByLabelText('Remove')).toHaveLength(3);`

Reply:
awesome! I was not aware of this. Thanks.